tests/src/test/scala/com/nvidia/spark/rapids/HashAggregatesSuite.scala

```diff
@@ -105,6 +105,29 @@ class HashAggregatesSuite extends SparkQueryCompareTestSuite {
       .agg(first(col("c0"), ignoreNulls = true), last(col("c0"), ignoreNulls = true))
   }
 
+  test("nullable aggregate with not null filter") {
+
+    withGpuSparkSession(spark => {
+      val df = firstDf(spark)
+          .coalesce(1)
+          .sort(col("c2").asc, col("c0").asc) // force deterministic use case
+          .groupBy(col("c2"))
+          .agg(min(col("c0")).alias("mymin"),
+            max(col("c0")).alias("mymax"))
+          .filter(col("mymin").isNotNull
+              .and(col("mymax").isNotNull))
+
+      df.collect()
+
+      val finalPlan = df.queryExecution.executedPlan
```

Comment:
nit: Do we actually need to run collect to get the executedPlan? I didn't think that was necessary.

Reply:
It would be if we were ever to switch to AQE being on by default. The other option would have been to use the capture callback. I can modify the test to do that.